test/OpenTelemetry.Instrumentation.Runtime.Tests/RuntimeMetricsTests.cs

```diff
@@ -34,9 +37,150 @@ public void RuntimeMetricsAreCaptured()
                  .AddInMemoryExporter(exportedItems)
                 .Build();
 
+            // The process.runtime.dotnet.exception.count metrics are only available after an exception has been thrown post OpenTelemetry.Instrumentation.Runtime initialization.
+            try
+            {
+                throw new Exception("Oops!");
+            }
+            catch (Exception)
+            {
+                // swallow the exception
+            }
+
             meterProvider.ForceFlush(MaxTimeToAllowForFlush);
             Assert.True(exportedItems.Count > 1);
             Assert.StartsWith(MetricPrefix, exportedItems[0].Name);
+
+            var assembliesCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.assemblies.count");
+            Assert.NotNull(assembliesCountMetric);
+
+            var exceptionsCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.exceptions.count");
+            Assert.True(GetValue(exceptionsCountMetric) >= 1);
+        }
+
+#if NET6_0_OR_GREATER
+        [Fact]
+        public void GcMetricsTest()
+        {
+            var exportedItems = new List<Metric>();
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                 .AddRuntimeInstrumentation()
+                 .AddInMemoryExporter(exportedItems)
+                .Build();
+
+            GC.Collect(1);
+
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+
+            var gcCountMetric = exportedItems.FirstOrDefault(i => i.Name == "process.runtime.dotnet.gc.collections.count");
+            Assert.NotNull(gcCountMetric);
+
+#if NETCOREAPP3_1_OR_GREATER
```

Comment:
this is already inside net6_0 or greater, right?

Reply:
That's right. I did a mistake by moving it and `gcCountMetric` to this method, which is all net6 or greater. I'll adjust the `NET6_0_OR_GREATER` to cover only the applicable metrics items.

Reply:
Done. I've also added a test for `gc.heap.fragmentation.size` on condition `if (Environment.Version.Major >= 7)`.